p2p/session.go

```diff
@@ -152,19 +170,32 @@ func (s *session[H]) doRequest(
 	req *p2p_pb.HeaderRequest,
 	headers chan []H,
 ) {
+	ctx, span := tracerSession.Start(ctx, "request-headers-from-peer", trace.WithAttributes(
+		attribute.String("peerID", stat.peerID.String()),
+		attribute.Int64("from", int64(req.GetOrigin())),
+		attribute.Int64("amount", int64(req.Amount)),
+	))
+
+	defer span.SetStatus(codes.Ok, "")
```

Comment:
Why do we always set it to ok? Should we handle errors via it?

Reply:
will fix it